rusoto/services/sts/src/custom/credential.rs

```diff
@@ -135,7 +135,7 @@ impl StsSessionCredentialsProvider {
                 token_code: self.mfa_code.clone(),
                 duration_seconds: Some(self.session_duration.num_seconds() as i64),
                 ..Default::default()
-            }) {
+            }).sync() {
```

Comment:
I'd prefer if we didn't use sync outside of tests (and maybe not even in tests). I know this is more work, but I think we should fully switch :smile:

Reply:
Just to make sure we're on the same page here, switching means changing the credentials crate to be fully async, including the `ProvideAwsCredentials` trait.

I'm happy to do that, it'd just mean a couple of days worth more work.